src/main/java/org/mule/module/json/internal/LibraryLink.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.json.internal;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * Defines the default behaviour to implement chain of responsibility design pattern
+ */
+public abstract class LibraryLink {
+
+  LibraryLink nextLink;
```

Comment:
Just in case, to avoid having the subclasses touching this reference, I would suggest to make this private.